api/rpc/stat/stat.go

```diff
@@ -1,44 +1,239 @@
 package stat
 
 import (
-	"time"
-
+	"fmt"
 	"github.com/appcelerator/amp/data/influx"
 	"golang.org/x/net/context"
+	//"time"
 )
 
 //Stat structure to implement StatServer interface
 type Stat struct {
-	conn influx.Influx
+	Influx influx.Influx
+}
+
+//CPUQuery Extract CPU information according to StatRequest
+func (s *Stat) CPUQuery(ctx context.Context, req *StatRequest) (*CPUReply, error) {
+	idFieldName, nameFieldName := getIDNameFields(req)
+	fmt.Println(idFieldName, nameFieldName )
+	query := buildQueryString(req, idFieldName, "cpu")
+	res, err := s.Influx.Query(query)
+	if err != nil {
+		return nil, err
+	}
+
+	cpuReply := CPUReply{}
+	list := res.Results[0].Series[0].Values
+	cpuReply.Entries = make([]*CPUEntry, len(list))
+	for i, row := range list {
+		entry, err := s.ressourceCPUQuery(ctx, req, idFieldName, nameFieldName, row[1].(string))
```

Comment:
I think we can improve on this. I do not like the idea of making a net call for each resource id obtained in the first query; this is inefficient. There is a way to accomplish this by executing 1 query using the group by and limit 1.  for example,
```
select 
"com.docker.swarm.service.name" as service, 
container_name, 
container_id, 
usage_in_kernelmode as usage_kernel, 
usage_in_usermode as usage_user, 
usage_total, 
usage_system 
from docker_container_cpu 
where time > now() - 1m 
GROUP BY "com.docker.swarm.service.name", container_name 
LIMIT 1
```
In the even that we should need to execute multiple queries in succession, then this approach is preferred:
https://docs.influxdata.com/influxdb/v0.13/query_language/data_exploration/#multiple-statements-in-queries